src/libostree/ostree-repo-commit.c

```diff
@@ -734,6 +734,17 @@ write_content_object (OstreeRepo         *self,
     }
   else
     {
+      /* Update size metadata if configured */
```

Comment:
Yeah, bare-user symlinks are the tricky case, for those we have `phys_object_is_symlink = FALSE, object_file_type = G_FILE_TYPE_SYMBOLIC_LINK`.  Although actually now that I look at this, the whole thing can only apply for `archive` repos:

```
      g_assert (repo_mode == OSTREE_REPO_MODE_ARCHIVE_Z2);

      if (self->generate_sizes)
        indexable = TRUE;
```

Anyways given the bare-user bits I think we should keep the conditional.